x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/ssl/SSLReloadDuringStartupIntegTests.java

```diff
@@ -59,66 +81,121 @@ protected boolean transportSSLEnabled() {
+            afterKeystoreFix.await(); // SYNC: Verify cluster after cert update
+        }
+        try (Timed t = new Timed(LOGGER, Level.INFO, "Ensure cluster size consistency took {}ms.")) {
+            ensureClusterSizeConsistency();
+        }
+        try (Timed t = new Timed(LOGGER, Level.INFO, "Ensure fully connected cluster took {}ms.")) {
+            ensureFullyConnectedCluster();
+        }
+    }
+
+    private void waitUntilNodeStartupIsReadyToBegin(final CountDownLatch beforeKeystoreFix) {
+        beforeKeystoreFix.countDown(); // SYNC: Cert update & ES restart
+        try {
+            final long sleepMillis = randomLongBetween(1L, 2000L); // intended sleepMillis
+            try (Timed t = new Timed(LOGGER, Level.INFO, "Awaited {}ms. Sleeping " + sleepMillis + "ms before fixing...")) {
+                beforeKeystoreFix.await(); // SYNC: Cert update & ES restart
+            }
+            try (Timed t = new Timed(LOGGER, Level.INFO, "Slept {}ms, intended " + sleepMillis + "ms. Fixing can start now...")) {
+                Thread.sleep(sleepMillis); // Simulate cert update delay relative to ES start
+            }
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
+    }
+
+    private void waitUntilFixKeystoreIsReadyToBegin(final CountDownLatch beforeKeystoreFix) {
+        beforeKeystoreFix.countDown(); // SYNC: Cert update & ES restart
+        try (Timed t = new Timed(LOGGER, Level.INFO, "Awaited {}ms. Node can start now...")) {
+            beforeKeystoreFix.await(); // SYNC: Cert update & ES restart
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
     }
 
-    private void copyAndAtomicMoveIfPossible(Path source, Path target) throws IOException {
+    /**
+     * Copy a source file to a destination file, overwriting if necessary.
+     * Use an intermediate temporary file, to attempt an atomic move.
+     * If atomic move fails, fall back to non-atomic move.
+     * @param source File to be copied.
+     * @param target File to be created or overwritten.
+     * @throws IOException Cannot create temp file, or copy source file to temp file, or non-atomic move temp file to target file.
+     */
+    private void atomicCopyIfPossible(Path source, Path target) throws IOException {
+        LOGGER.info("Copying [{}] to [{}]", source, target);
         Path tmp = createTempFile();
+        // LOGGER.info("Temporary file [{}] created", tmp);
         Files.copy(source, tmp, StandardCopyOption.REPLACE_EXISTING);
+        // LOGGER.info("Copy succeeded from [{}] to [{}]", source, tmp);
         try {
             Files.move(tmp, target, StandardCopyOption.REPLACE_EXISTING, StandardCopyOption.ATOMIC_MOVE);
+            LOGGER.info("Atomic move succeeded from [{}] to [{}]", tmp, target);
         } catch (AtomicMoveNotSupportedException e) {
+            LOGGER.warn("Atomic move failed from [{}] to [{}]", tmp, target, e);
```

Comment:
nit: since these are developer level log messages a few of these without timings seem unnecessary since the person reading the log messages will also be in the code. (here we can infer it failed if Non-atomic message is printed and the exception probably isn't relevant). Not a big deal, just trying to keep visual clutter down.

Reply:
Removed